git_dumper.py

```diff
@@ -459,10 +478,18 @@ def fetch_git(url, directory, jobs, retry, timeout, http_headers, client_cert_p1
             jobs,
             args=(url, directory, retry, timeout, http_headers),
         )
+        
+        os.chdir(directory)
+
+        printf("[-] Sanitizing .git/config\n")
+        sanitize_file(".git/config")
 
         printf("[-] Running git checkout .\n")
-        os.chdir(directory)
-        subprocess.check_call(["git", "checkout", "."])
+        ENV = os.environ
+        configured_proxy = socks.getdefaultproxy()
+        if configured_proxy[0] > 0:
```

Comment:
What happens if there are no proxy configured? It looks like `configured_proxy` would be None.

Reply:
Sorry, it's indeed a None type, so changing 490 line to:

```python
if configured_proxy != None:
```

should not raise exception in case of no proxy is set. I thought I tested a no proxy usage but looks like I didn't.